src/controllers/slot.ts

```diff
@@ -12,105 +10,179 @@ function isSlotElement(node: Element): node is HTMLSlotElement {
+   * @return {boolean}
+   */
+  public has(slot: string | Ref<HTMLSlotElement>, selector?: string): boolean {
+    const slotName = typeof slot === 'string' ? slot : slot.value?.name;
+
+    this.__hasCalled = true;
+
+    if (!slotName) {
+      return false;
+    }
+
+    const slots = this.__slots.filter((slot) => {
+      return (
+        slotName === '*' ||
+        (!slot.hasAttribute('name') && slotName === 'default') ||
+        slot.name === slotName
+      );
+    });
+    const children = slots.reduce<Element[]>((accum, slot) => {
+      for (const child of slot.assignedElements()) {
+        accum.push(child);
+      }
+      return accum;
+    }, []);
+
+    if (!selector) {
+      return children.length > 0;
+    }
+
+    return children.some((child) => child.matches(selector));
+  }
 
   /**
    * Binds a function to be called any time elements matching the specified
-   * selector are slotted into the slot
+   * selector are slotted into a given slot
+   * @param {string|Ref} slot Slot to observe, * for all slots
    * @param {string} selector Query selector to filter elements by. Set this
    * to `*` for all elements
    * @param {SlotListenerCallback} callback Callback to call
    * @return {void}
    */
   public addListener<T extends keyof HTMLElementTagNameMap>(
+    slot: string | Ref<HTMLSlotElement>,
     selector: T,
     callback: SlotListenerCallback<HTMLElementTagNameMap[T]>
   ): void;
 
   /** @inheritdoc */
   public addListener(
+    slot: string | Ref<HTMLSlotElement>,
     selector: string,
     callback: SlotListenerCallback<Element>
   ): void;
 
   /** @inheritdoc */
   public addListener(
+    slot: string | Ref<HTMLSlotElement>,
     selector: string,
     callback: SlotListenerCallback<Element>
   ): void {
     this.__listeners.add({
+      slot,
       selector,
       callback
     });
   }
 
   /**
    * Fired when a slot change event occurred
+   * @param {Event} ev Event fired
    * @return {void}
    */
-  private __onSlotChanged = (): void => {
-    const element = this._element;
+  private __onSlotChanged = (ev: Event): void => {
+    const element = ev.target as Element;
 
-    if (!element || !isSlotElement(element)) {
+    if (!isSlotElement(element)) {
       return;
     }
 
     const nodes = element.assignedElements();
 
     for (const listener of this.__listeners) {
+      if (listener.slot !== '*' && !slotMatches(element, listener.slot)) {
+        continue;
+      }
+
       for (const node of nodes) {
         if (listener.selector === '*' || node.matches(listener.selector)) {
           listener.callback(node);
         }
       }
     }
-  };
 
-  /** @inheritdoc */
-  protected override _onElementChanged(prevElement: Element | undefined): void {
-    super._onElementChanged(prevElement);
-
-    if (prevElement) {
-      prevElement.removeEventListener('slotchange', this.__onSlotChanged);
+    if (this.__hasCalled) {
+      this.__host.requestUpdate();
```

Comment:
WOWOWOWOWOWOWOWOW 💯 💯 💯 

Only concern, do we care about this case (no is perfectly valid, and I'm leaning towards that):

```ts
render() {
  html`${this.booleanThatStartsOffAsTrueButThenFalseForTheMajorityOfTime ?  this.ctrl.has(...) : ''}`
}
```

because that would call request update even though that `.has()` doesn't change the render output